src/sgame/components/MissileComponent.cpp

```diff
@@ -83,13 +83,14 @@ static trace2_t MissileTrace(gentity_t* ent)
 
 		if (trWorld.startsolid)
 		{
-			return trWorld;
+			result = trWorld;
+		}
+		else
+		{
+			trace2_t trBody = G_Trace2(ent->r.currentOrigin, ent->r.mins, ent->r.maxs, trWorld.endpos,
```

Comment:
The first trace goes the entire length from `ent->r.currentOrigin` to `origin`. The second trace goes only up to `trWorld.endpos`, making it shorter in the bugged case. Thus the two trace fractions can't be directly compared and the ternary below is broken.

Changing the endpoint of this trace to `origin` fixes the issue.

What is the purpose of `pointAgainstWorld` in the first place?

Reply:
> Changing the endpoint of this trace to `origin` fixes the issue.

Sure, making both traces use the same endpoint would be another way to fix it.

> What is the purpose of `pointAgainstWorld` in the first place?

It makes the missile use a point trace against map geometry (CONTENTS_SOLID) but a finite sized box trace against CONTENTS_BODY entities.

Reply:
> It makes the missile use a point trace against map geometry (CONTENTS_SOLID) but a finite sized box trace against CONTENTS_BODY entities.

Oh, that makes sense, thanks.

> Sure, making both traces use the same endpoint would be another way to fix it.

It's _the_ fix. Comparing fractions from completely different traces doesn't make any sense. Your code is still bugged.

Reply:
With the proposed change there is no longer a comparison of two fractions. Only the comparison of a fraction with 1.0.

Reply:
You're right, my bad.

I tested your branch (rebased on top of `v0.55.3`) and everything works fine. I can't reproduce Reaper's issue.